deployment/common/changeset/save_existing_test.go

```diff
@@ -47,8 +48,92 @@ func TestSaveExisting(t *testing.T) {
 	require.NoError(t, err)
 	require.NoError(t, dummyEnv.ExistingAddresses.Merge(output.AddressBook))
 	addresses, err := dummyEnv.ExistingAddresses.Addresses()
+	require.NoError(t, err)
 	require.Len(t, addresses, 2)
 	addressForChain1, exists := addresses[chainsel.TEST_90000001.Selector]
 	require.True(t, exists)
 	require.Len(t, addressForChain1, 1)
 }
+
+func TestSaveExistingAddressWithLabels(t *testing.T) {
+	dummyEnv := deployment.Environment{
+		Name:              "dummy",
+		Logger:            logger.TestLogger(t),
+		ExistingAddresses: deployment.NewMemoryAddressBook(),
+		Chains: map[uint64]deployment.Chain{
+			chainsel.TEST_90000001.Selector: {},
+			chainsel.TEST_90000002.Selector: {},
+		},
+	}
+	dummyType1 := deployment.TypeAndVersion{
+		Type:    "dummyType",
+		Version: deployment.Version1_5_0,
+	}
+	dummyType1.AddLabel("label1")
+	dummyType1.AddLabel("label2")
+	ExistingContracts := ExistingContractsConfig{
+		ExistingContracts: []Contract{
+			{
+				Address:        common.BigToAddress(big.NewInt(1)).String(),
+				TypeAndVersion: dummyType1,
+				ChainSelector:  chainsel.TEST_90000001.Selector,
+			},
+		},
+	}
+
+	output, err := SaveExistingContractsChangeset(dummyEnv, ExistingContracts)
+	require.NoError(t, err)
+	require.NoError(t, dummyEnv.ExistingAddresses.Merge(output.AddressBook))
+	addresses, err := dummyEnv.ExistingAddresses.Addresses()
+	require.NoError(t, err)
+	require.Len(t, addresses, 1)
+	addressForChain1, exists := addresses[chainsel.TEST_90000001.Selector]
+	require.True(t, exists)
+	require.Len(t, addressForChain1, 1)
+	require.Equal(t, "dummyType 1.5.0 label1 label2", addressForChain1[common.BigToAddress(big.NewInt(1)).String()].String())
```

Comment:
Nit: This test is going to be brittle, but it's probably fine. asserting on string representations isn't ideal. Might be better to make some convenience constructors for easy composition of typeandversions in testing. But like, it's a small thing. Feel free to do it as a follow-up.